bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/BlueZBluetoothDevice.java

```diff
@@ -390,10 +416,54 @@ private BluetoothGattDescriptor getTinybDescriptorByUUID(String uuid) {
         return null;
     }
 
+    private void cancelInactiveCleanupJob() {
+        try {
+            inactiveCleanupJobLock.lock();
+            if (inactiveCleanupJob != null) {
+                inactiveCleanupJob.cancel(true);
+                inactiveCleanupJob = null;
+            }
+        } finally {
+            inactiveCleanupJobLock.unlock();
+        }
+    }
+
+    private void onActivity() {
+        logger.trace("device scanned: {}", device.getAddress());
+        try {
+            inactiveCleanupJobLock.lock();
+            cancelInactiveCleanupJob();
+            inactiveCleanupJob = scheduler.schedule(() -> {
+                synchronized (BlueZBluetoothDevice.this) {
+                    if (device != null && !device.getConnected()) {
+                        logger.debug("Removing device '{}' due to inactivity", device.getAddress());
+                        try {
+                            disableNotifications();
+                            device.remove();
+                        } catch (BluetoothException ex) {
+                            if (ex.getMessage().contains("Does Not Exist")) {
```

Comment:
Can't you separate by Exception class? That is what the classes should normally be used for, this is really a last resort.

Reply:
I would if I could. BluetoothException is what they wrap all the underlying jni exceptions with. Other parts of the code were already using this pattern so I just decided to follow it.